packages/shared/src/components/post/PostContentHeader.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { ReactElement } from 'react';
+import { useRouter } from 'next/router';
+import { Button, ButtonSize, ButtonVariant } from '../buttons/Button';
+import { ArrowIcon } from '../icons';
+import { PostHeaderActions } from './PostHeaderActions';
+import { PostNavigationProps } from './common';
+import { SourceType } from '../../graphql/sources';
+
+export function PostContentHeader({
+  post,
+  onReadArticle,
+}: Pick<PostNavigationProps, 'onReadArticle' | 'post'>): ReactElement {
+  const router = useRouter();
+  const canGoBack = !!globalThis?.window?.history?.length;
+  const routedFromSquad = router?.query?.squad;
+  const squadLink = `/squads/${router.query.squad}`;
+
+  if (!canGoBack && !routedFromSquad && post.source.type !== SourceType.Squad) {
+    return null;
+  }
+
+  const onBackClick = () => {
```

Comment:
the conditions here seem to be related only to squad post types or routed from squads else it returns null, should it just go back to history if there is history?

Reply:
Nvm saw the other condition below, though think is best if all conditions are here

Reply:
Yeah, that makes sense. Let's do that.